internal/verifier/partition.go

```diff
@@ -225,24 +197,56 @@ func (verifier *Verifier) createPartitionTasksWithSampleRateRetryable(
 		return nil
 	}
 
-	lowerBound := lowerBoundOpt.OrElse(primitive.MinKey{})
+	idealNumPartitions := util.Divide(collBytes, idealPartitionBytes)
+	docsPerPartition := util.Divide(docsCount, idealNumPartitions)
```

Comment:
It seems that it's still possible to get division by 0 error. If `collBytes  < idealPartitionBytes`, `idealNumPartitions` would become 0. Could we limit `idealNumPartitions` and `docsPerPartition` to be larger than 1?
```suggestion
	idealNumPartitions := min(1, util.Divide(collBytes, idealPartitionBytes))
	docsPerPartition := min(1, util.Divide(docsCount, idealNumPartitions))
```

Reply:
util.Divide() returns a float, so that shouldn’t happen, but I’ve tweaked the logic anyhow.